Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.cpp

```diff
@@ -57,17 +74,11 @@ namespace LuaBuilder
                         asset.SetAutoLoadBehavior(AZ::Data::AssetLoadBehavior::PreLoad);
                         assets.push_back(asset);
                     }
-                    else if(isLuaDependency)
-                    {
-                        AZ_Error("LuaBuilder", false, "Did not find dependency %s referenced by script.", dependency.m_dependencyPath.c_str());
-                    }
                     else
                     {
-                        AZ_Error("LuaBuilder", false, "%s referenced by script does not appear to be a lua file.\n"
-                            "References to assets should be handled using Property slots and AssetIds to ensure proper dependency tracking.\n"
-                            "This file will not be tracked as a dependency.",
-                            dependency.m_dependencyPath.c_str());
+                        AZ_Error("LuaBuilder", false, "Invalid asset %s referenced by script.", dependency.m_dependencyPath.c_str());
```

Comment:
It makes sense to mention that the asset is not recognized, there might be an asset type that is not supported by the asset map yet. So there should be a hint to developer on fixing the issue.

Reply:
Added a clarification line